agroal-api/src/main/java/io/agroal/api/cache/ConnectionCache.java

```diff
@@ -0,0 +1,53 @@
+// Copyright (C) 2017 Red Hat, Inc. and individual contributors as indicated by the @author tags.
+// You may not use this file except in compliance with the Apache License, Version 2.0.
+
+package io.agroal.api.cache;
+
+/**
+ * Interface for a cache of connections. It's intended for mapping connections to executing threads.
+ *
+ * @author <a href="lbarreiro@redhat.com">Luis Barreiro</a>
+ */
+public interface ConnectionCache {
```

Comment:
Shouldn't this have <T extends Acquirable> and get/put take type T so there is no need to cast?

Reply:
There will always be a need to cast, as the `ConnectionHandler` class is not on the API.